### PR TITLE
[#4657] Do not set bulk flag with force flag (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
CI tests running

### CODE REVIEW

**Comment by trel** [comment]
seems good - if the tests are happy, please to #

File: scripts/irods/test/test_iput_options.py [hunk 1]
```diff
@@ -15,6 +16,7 @@
 from ..configuration import IrodsConfig
 from .. import lib
 from . import session
+from . import ustrings
```
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [redefinition of unused 'ustrings' from line 6](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

File: scripts/irods/test/test_iput_options.py [hunk 2]
```diff
@@ -195,6 +197,33 @@ def test_iput_symlink_to_self_on_commandline_4016(self):
             self.user0.assert_icommand_fail( cmd, 'STDOUT_SINGLELINE', symtoself );
             self.user0.assert_icommand_fail( cmd, 'STDOUT_SINGLELINE', goodfile );
 
+    def test_iput_recursive_bulk_upload__4657(self):
+        bulk_str = 'Bulk upload'
+        source_path = tempfile.mkdtemp()
+        single_file = tempfile.mkstemp(dir=source_path)
```
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [Unused variable 'single_file'](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [local variable 'single_file' is assigned to but never used](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

File: scripts/irods/test/test_iput_options.py (hunk 1 above)
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [Reimport 'ustrings' (imported line 6)](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

File: scripts/irods/test/test_iput_options.py [hunk 3]
```diff
@@ -195,6 +197,33 @@ def test_iput_symlink_to_self_on_commandline_4016(self):
...
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-f', source_path, 'v-r-f'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertNotIn(bulk_str, out, 'Bulk upload performed on 1 file')
+        
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-b', source_path, 'v-r-b'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertIn(bulk_str, out, 'Bulk upload not performed when requested')
+
```
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

File: scripts/irods/test/test_iput_options.py [hunk 4]
```diff
@@ -195,6 +197,33 @@ def test_iput_symlink_to_self_on_commandline_4016(self):
...
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-b', source_path, 'v-r-b'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertIn(bulk_str, out, 'Bulk upload not performed when requested')
+        
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-b', '-f', source_path, 'v-r-b-f'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertIn(bulk_str, out, 'Bulk upload not performed when requested')
+        finally:
```
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

File: scripts/irods/test/test_iput_options.py [hunk 5]
```diff
@@ -195,6 +196,33 @@ def test_iput_symlink_to_self_on_commandline_4016(self):
...
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-b', source_path, 'v-r-b'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertIn(bulk_str, out, 'Bulk upload not performed when requested')
+
+            _,out,_ = self.admin.assert_icommand(
+                ['iput', '-v', '-r', '-b', '-f', source_path, 'v-r-b-f'],
+                'STDOUT', ustrings.recurse_ok_string())
+            self.assertIn(bulk_str, out, 'Bulk upload not performed when requested')
+        finally:
```
Comment (trel) [inline scripts/irods/test/test_iput_options.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893065)

**Comment by alanking** [comment]
Tests happy.

**Pull request merged**